From d9043cece2c04905bcdfea6c600651fd8e4e0be6 Mon Sep 17 00:00:00 2001
From: Benjamin Marzinski <bmarzins@sourceware.org>
Date: Fri, 2 Oct 2009 22:01:24 +0200
Subject: [PATCH] [kpartx] make kpartx deal with more than 256 minor numbers

Fix for bz #526550.  Fix kpartx MAKEDEV macro so it can deal with more
than 256 minor numbers.
---
 kpartx/devmapper.c |    1 -
 kpartx/devmapper.h |    4 ++++
 kpartx/kpartx.c    |    1 -
 3 files changed, 4 insertions(+), 2 deletions(-)

diff -Nurp multipath-tools-0.4.8.orig/kpartx/devmapper.c multipath-tools-0.4.8/kpartx/devmapper.c
--- multipath-tools-0.4.8.orig/kpartx/devmapper.c	2007-08-03 00:05:37.000000000 +0300
+++ multipath-tools-0.4.8/kpartx/devmapper.c	2010-02-20 19:48:11.239420130 +0200
@@ -6,7 +6,6 @@
 #include <string.h>
 #include <libdevmapper.h>
 #include <ctype.h>
-#include <linux/kdev_t.h>
 #include <errno.h>
 
 #define UUID_PREFIX "part%d-"
diff -Nurp multipath-tools-0.4.8.orig/kpartx/devmapper.h multipath-tools-0.4.8/kpartx/devmapper.h
--- multipath-tools-0.4.8.orig/kpartx/devmapper.h	2007-08-03 00:05:37.000000000 +0300
+++ multipath-tools-0.4.8/kpartx/devmapper.h	2010-02-20 19:48:11.240420018 +0200
@@ -1,3 +1,7 @@
+#define MAJOR(dev)      ((dev & 0xfff00) >> 8)
+#define MINOR(dev)      ((dev & 0xff) | ((dev >> 12) & 0xfff00))
+#define MKDEV(ma,mi)    ((mi & 0xff) | (ma << 8) | ((mi & ~0xff) << 12))
+
 int dm_prereq (char *, int, int, int);
 int dm_simplecmd (int, const char *);
 int dm_addmap (int, const char *, const char *, const char *, unsigned long,
diff -Nurp multipath-tools-0.4.8.orig/kpartx/kpartx.c multipath-tools-0.4.8/kpartx/kpartx.c
--- multipath-tools-0.4.8.orig/kpartx/kpartx.c	2010-02-20 19:46:30.000000000 +0200
+++ multipath-tools-0.4.8/kpartx/kpartx.c	2010-02-20 19:48:11.240420018 +0200
@@ -29,7 +29,6 @@
 #include <sys/types.h>
 #include <ctype.h>
 #include <libdevmapper.h>
-#include <linux/kdev_t.h>
 
 #include "devmapper.h"
 #include "crc32.h"
